gson/src/main/java/com/google/gson/FieldNamingPolicy.java

```diff
@@ -149,10 +150,6 @@ static String upperCaseFirstLetter(String name) {
       firstCharacter = name.charAt(++index);
     }
 
-    if (index == name.length()) {
```

Comment:
I wanted to leverage this behavior for my LOWER_CAMEL_CASE fieldNamingPolicy, so before refactoring, I surrounded it with unit tests.  (I'm a unit test junkie)  I did an emma code coverage test, and realized it was impossible for me to put anything into any of my unit tests.  (The pit mutation test also flagged it as unreadable with my existing permutations) 

That said, understandable on being concerned with such a highly used class, so if you want, I'll put it back in there.  (Although, I was hoping my unit tests could mitigate the risk).  Let me know how you want to proceed, and I'll commit those changes.